### PR TITLE
Fixed #121 So the viewbox is scaled

### LINKED ISSUE #121: When using viewBox attribute, canvas size is not scaled to width and height of svg attribute
When trying to convert the following svg to png

```
from svglib.svglib import svg2rlg
from reportlab.graphics import renderPDF, renderPM

svgtemplate="""
<svg   width="1250" height="1250"  viewBox="-40 -40 80 80">
<rect width="30" height="30" style="fill:red" />
</svg>
"""
def png(svg,pngfilename):
    text_file = open("template.svg", "w")
    text_file.write(svg)
    text_file.close()
    drawing = svg2rlg("template.svg")
    renderPM.drawToFile(drawing, pngfilename)

png(svgtemplate,'Test_viewBox.png')
```

I receive a png that is only 80px x 80px.
It should be 1250 x 1250 and the rect should be scaled accordingly.

(svglib 0.8.1 python 3.6.4 Windows 7)

### PR DESCRIPTION
**Pull request by tomturner**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,5 +14,8 @@ docs/_build
 *.swp
 *.egg
 env/
+venv/
 .workon
 dist/
+.idea/
+.DS_Store
```
Comment (claudep) [inline .gitignore]: @deeplook may have a different opinion, but as for me, those user-specific or platform-specific things should be excluded in users global gitignores (I'd prefer removing some of existing lines than continue to add more).

File: svglib/svglib.py [hunk 2]
```diff
@@ -284,9 +284,11 @@ def convertLength(self, svgAttr, percentOf=100, em_base=12):
             return float(text[:-2]) * em_base
         elif text.endswith("px"):
             return float(text[:-2])
+        elif text.endswith("mm"):
+            return float(text[:-2]) * mm
```
Comment (claudep) [inline svglib/svglib.py]: Nice improvement, but this should happen in a separate pull request, with a test. Would you be able to do that please?

File: svglib/svglib.py [hunk 3]
```diff
@@ -425,15 +427,18 @@ def __init__(self, path=None,color_converter=None):
...
         for xlink in self.waiting_use_nodes.keys():
             logger.debug("Ignoring unavailable object width ID '%s'." % xlink)
-
-        main_group.scale(1, -1)
+        scale_x = self.width / self.box.width
+        scale_y = self.height / self.box.height
+        main_group.scale(scale_x, -scale_y)
         main_group.translate(0 - self.box.x, -self.box.height - self.box.y)
-        drawing = Drawing(self.box.width, self.box.height)
+        drawing = Drawing(self.width, self.height)
```
Comment (claudep) [inline svglib/svglib.py]: We are about to commit support for embedded SVGs in #124, which will heavily touch this part of the code. It would be great to wait for that other PR and then try to rebase yours onto the new code.

File: svglib/svglib.py [hunk 4]
```diff
@@ -801,7 +808,7 @@ def convertPath(self, node):
         subpath_start = []
         lastop = ''
 
-        for i in xrange(0, len(normPath), 2):
+        for i in range(0, len(normPath), 2):
```
Comment (claudep) [inline svglib/svglib.py]: It's indeed strange that `xrange` usage doesn't crash in current tests, as it's not present by default in Python 3. I guess reportlab imports are doing some magics to alias it. I'm in favor of the change, but this should also be part of another PR with the other usage of `xrange` in svglib.

File: .gitignore (hunk 1 above)
Comment (tomturner) [reply .gitignore]: Ok far enough. The fist one was PYCharm the second was a file that get generated on a Mac

File: svglib/svglib.py (hunk 4 above)
Comment (tomturner) [reply svglib/svglib.py]: Done in #130

File: svglib/svglib.py (hunk 3 above)
Comment (tomturner) [reply svglib/svglib.py]: Let me know when you want me to do this.

File: svglib/svglib.py (hunk 2 above)
Comment (tomturner) [reply svglib/svglib.py]: Done in #131

Comment (deeplook) [reply svglib/svglib.py]: I've commented there about some simple test code that would make this nicely complete. Otherwise thanks!

File: .gitignore (hunk 1 above)
Comment (deeplook) [reply .gitignore]: Yep, these should surely not go into the repo.

File: svglib/svglib.py (hunk 3 above)
Comment (claudep) [reply svglib/svglib.py]: #124 has been merged, so you can try to rebase your patch.

Comment (claudep) [reply svglib/svglib.py]: Oh, and if you have an example (SVG + expected result), that would be great.

Comment (tomturner) [reply svglib/svglib.py]: Ok I will write the patch again and send it in

File: svglib/svglib.py (hunk 4 above)
Comment (deeplook) [reply svglib/svglib.py:811]: Sorry, but is this still relevant? I can't figure out why it got stalled for so long.

**Comment by claudep** [comment]
As the original ticket is now closed, I guess this PR can be closed also.
